metrics/scope.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 GRAIL, Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package metrics
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"sync/atomic"
+	"unsafe"
+)
+
+// Scope is a collection of metric instances.
+type Scope struct {
+	storage unsafe.Pointer // stores *[]interface{}
+}
+
+// GobEncode implements a custom gob encoder for scopes.
+func (s *Scope) GobEncode() ([]byte, error) {
+	var b bytes.Buffer
```

Comment:
you could consider sync.Pool if you expect these bytes.Buffers' to be rapidly recycled, eg. when merging scopes?

Reply:
they're only used when transmitting scopes across the wire after task completion... i'd rather not introduce manual memory management until there's proof in the pudding (profiles)